tests/remote_resource/test_requests.py

```diff
@@ -156,9 +206,38 @@ def test_correct_resource_type(self, mock_zc_settings, mock_requests):
         mock_requests.get.assert_called_once_with(self.gateway_root_path)
         self.assertEqual(expected_url, result_url)
 
-    def test_missing_required_parameters(self, mock_zc_settings, mock_requests):
-        resource_type = 'User'
-        self.assertRaises(Exception, get_route_from_fk, resource_type)
+    def test_correct_resource_type_with_one_query_parameter(self, mock_zc_settings, mock_requests):
+        self.bind_mock_objects(mock_zc_settings, mock_requests)
+        expected_url = 'https://mp-vendors.zerocater.com/vendors?filter[id__in]=[1,2,3]'
+
+        result_url = get_route_from_fk('Vendor', query_params={'filter[id__in]': '[1,2,3]'})
+
+        mock_requests.get.assert_called_once_with(self.gateway_root_path)
+        self.assertEqual(expected_url, result_url)
+
+    def test_correct_resource_type_with_more_query_parameters(self, mock_zc_settings, mock_requests):
+        self.bind_mock_objects(mock_zc_settings, mock_requests)
+        expected_url_1 = 'https://mp-vendors.zerocater.com/vendors?filter[id__in]=[1,2,3]&name=testing'
+        expected_url_2 = 'https://mp-vendors.zerocater.com/vendors?name=testing&filter[id__in]=[1,2,3]'
+
+        result_url = get_route_from_fk('Vendor', query_params={'filter[id__in]': '[1,2,3]', 'name': 'testing'})
+
+        mock_requests.get.assert_called_once_with(self.gateway_root_path)
+        self.assertIn(result_url, [expected_url_1, expected_url_2])
+
+    def test_correct_resource_type_with_all_parameters(self, mock_zc_settings, mock_requests):
+        self.bind_mock_objects(mock_zc_settings, mock_requests)
+        expected_url = 'https://mp-vendors.zerocater.com/vendors/1?filter[id__in]=[1,2,3]'
```

Comment:
This URL shouldn't ever be generated. If someone were to request Vendor 1, they would get `https://mp-vendors.zerocater.com/vendors/1` but if they were to request Vendor 1, 2, and 3 they would get `https://mp-vendors.zerocater.com/vendors?filter[id__in]=1,2,3`